### PR TITLE
Use unary requests in the echo example

### PR DESCRIPTION
**Pull request by jul-sh**

### CODE REVIEW

File: oak_functions/client/rust/src/lib.rs [hunk 1]
```diff
@@ -48,14 +63,35 @@ impl Client {
...
-            .and_then(|rsp| Response::decode(rsp.as_ref()).context("Could not decode the response"))
+}
+
+#[async_trait]
+impl Session for Client {
+    async fn invoke(&mut self, request: Request) -> anyhow::Result<Response> {
+        handle_response(self.inner.send(request).await)
+    }
+}
+
```
Comment (jul-sh) [inline oak_functions/client/rust/src/lib.rs:74]: This doesn't define the `new` method (despite the shared type signature), as the compiler warns that not all futures involved implement the `Send` trait (which the async-trait macro requires by default. 

async-trait does offer a [non-thread safe option](https://crates.io/crates/async-trait), but not sure if it's safe to use. Probably fine to keep it like though.

Comment (tiziano88) [reply oak_functions/client/rust/src/lib.rs:74]: Yeah the current version seems fine, also I believe there is never a case in which the new method needs to be invoked from a dyn trait object, it's always called on the concrete type, so lifting it into this trait would probably not be appropriate anyways.

**Review by tiziano88 — APPROVED** [review]

File: FORCE_CI [hunk 2]
```diff
@@ -1 +1 @@
-8
+9
```
[the remark this answers is not in the document]
Comment (jul-sh) [reply FORCE_CI:1]: image

i'm just the messenger 😅

File: oak_functions/examples/echo/example.toml [hunk 3]
```diff
@@ -15,5 +15,6 @@ additional_args = ["--config-path=./oak_functions/examples/echo/config.toml"]
 rust = { Cargo = { cargo_manifest = "oak_functions/client/rust/Cargo.toml" }, additional_args = [
   "--uri=http://localhost:8080",
   "--request=test",
-  "--expected-response-pattern=test"
+  "--expected-response-pattern=test",
+  "--unary-request-model"
```
[the remark this answers is not in the document]
Comment (jul-sh) [reply oak_functions/examples/echo/example.toml:19]: idk, ideally compile time flag would be nicest, but will do for now. :)

File: oak_functions/client/rust/src/lib.rs (hunk 1 above)
Comment (jul-sh) [reply oak_functions/client/rust/src/lib.rs:74]: Would potentially make the main.rs slightly nicer, but agree not strictly needed

**Comment by jul-sh** [comment]
@tiziano88 thx for the review!

**Pull request merged**